lib/class-wp-block-type-registry.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Blocks API: WP_Block_Type_Registry class
+ *
+ * @package gutenberg
+ * @since 0.2.0
+ */
+
+/**
+ * Core class used for interacting with block types.
+ *
+ * @since 0.2.0
+ */
+final class WP_Block_Type_Registry {
+	/**
+	 * Registered block types, as `$name => $instance` pairs.
+	 *
+	 * @since 0.2.0
+	 * @access private
+	 * @var array
+	 */
+	private $registered_block_types = array();
+
+	/**
+	 * Container for the main instance of the class.
+	 *
+	 * @since 0.2.0
+	 * @access private
+	 * @static
+	 * @var WP_Block_Type_Registry|null
+	 */
+	private static $instance = null;
```

Comment:
I think testing should be possible as we could manually spin up instances of the `WP_Block_Type_Registry` class for most test cases, as we wouldn't need to use the main instance. Once tests have been added, we'll know for sure.